apps/demo/src/app/service/custom-contact.ts

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+import { Contact, ContactSubscription, CustomContactFactory } from '@pazznetwork/ngx-chat-shared';
+import { dummyAvatar } from './dummy-avatar';
+
+export class CustomContact implements CustomContactFactory {
+  create(
+    jid: string,
+    name: string,
+    avatar: string = dummyAvatar,
+    subscription = ContactSubscription.none
+  ): Promise<Contact> {
+    return new Promise((resolve) =>
+      setTimeout(() => resolve(new Contact(jid, name, avatar, subscription)), 10)
```

Comment:
we wanted to simulate a real promise, with new Promise() the promise resolves directly and doesn't go to the event loop. Because of this we had a race condition on our project where we integrate the chat but not in the demo project.

Reply:
no i mean write a comment so future devs understand it

Reply:
ah yeah... makes sense lol